enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -957,19 +962,27 @@ func (c *podmanCommandContainer) killContainerIfRunning(ctx context.Context) err
 	return err
 }
 
-// An image can be corrupted if "podman pull" command is killed when pulling a parent layer.
-// More details can be found at https://github.com/containers/storage/issues/1136. When this
-// happens when need to remove the image before re-pulling the image in order to fix it.
+// An image can be corrupted in a couple of circumstances:
+//   - If "podman pull" command is killed when pulling a parent layer, more details:
+//     https://github.com/containers/storage/issues/1136
+//   - Due to a bug streaming the container image:
+//     https://github.com/buildbuddy-io/buildbuddy-internal/issues/2570
+//
+// In either case we need to remove the image before re-pulling the image in order to fix it.
 func (c *podmanCommandContainer) maybeCleanupCorruptedImages(ctx context.Context, result *interfaces.CommandResult) error {
-	if result.ExitCode != podmanInternalExitCode {
-		return nil
+	stderr := string(result.Stderr)
+	if result.ExitCode == podmanInternalExitCode && storageErrorRegex.MatchString(stderr) {
+		log.Debug("podman detected storage corruption, removing image")
+		result.Error = status.UnavailableError("a storage corruption occurred")
+		result.ExitCode = commandutil.NoExitCode
+		return removeImage(ctx, c.image)
+	} else if result.ExitCode == podmanCannotInvokeExitCode && inputOutputErrorRegex.MatchString(stderr) {
+		log.Debug("podman detected input/output error, removing image")
```

Comment:
nit: I think these are worth logging at info level so we can tell how often we're hitting these in prod

also probably worth using `Ctx` funcs so we can associate the logs with invocation IDs

```suggestion
		log.CtxInfo(ctx, "podman detected storage corruption, removing image")
		result.Error = status.UnavailableError("a storage corruption occurred")
		result.ExitCode = commandutil.NoExitCode
		return removeImage(ctx, c.image)
	} else if result.ExitCode == podmanCannotInvokeExitCode && inputOutputErrorRegex.MatchString(stderr) {
		log.CtxInfo(ctx, "podman detected input/output error, removing image")
```